interface/main/calendar/index.php

```diff
@@ -37,25 +37,32 @@
 require_once 'includes/pnAPI.php';
 require_once("$srcdir/acl.inc");
 
+
+
 // From Michael Brinson 2006-09-19:
 if (isset($_POST['pc_username'])) {
-    $_SESSION['pc_username'] = $_POST['pc_username'];
+    $pc_username=$_POST['pc_username'];
+    $pc_username[0]=filter_var( $pc_username[0],FILTER_SANITIZE_STRING);
+    $_SESSION['pc_username'] = $pc_username;
 }
 
 //(CHEMED) Facility filter
 if (isset($_POST['all_users'])) {
-    $_SESSION['pc_username'] = $_POST['all_users'];
+    $_SESSION['pc_username'] = $_POST['all_users'];;
 }
 
 // bug fix to allow default selection of a provider
 // added 'if..POST' check -- JRM
 if (isset($_REQUEST['pc_username']) && $_REQUEST['pc_username']) {
-    $_SESSION['pc_username'] = $_REQUEST['pc_username'];
+    $pc_username=$_REQUEST['pc_username'];
+    $pc_username[0]=filter_var( $pc_username[0],FILTER_SANITIZE_STRING);
+    $_SESSION['pc_username'] = $pc_username;
 }
 
 // (CHEMED) Get the width of vieport
 if (isset($_GET['framewidth'])) {
-    $_SESSION['pc_framewidth'] = $_GET['framewidth'];
+    $framewidth=filter_var($_GET['framewidth'],FILTER_SANITIZE_NUMBER_FLOAT);
```

Comment:
If this is always a number then what we have been doing is simply adding 0 to it:
``````
$framewidth= $_GET['framewidth'] + 0;

Reply:
I checked several scenarios and indeed framewidth is a number 
I don't know why this line exist , do you want me to delete it? 
![image](https://user-images.githubusercontent.com/20621444/32829912-316c446e-c9fc-11e7-89c2-344430358206.png)

Reply:
T think could just change the original line to (and remove the 2 lines you added):
``````
$_SESSION['pc_framewidth'] = $_GET['framewidth'] + 0;